fuel/datasets/cifar10.py

```diff
@@ -33,23 +34,12 @@ class CIFAR10(H5PYDataset):
     filename = 'cifar10.hdf5'
 
     def __init__(self, which_set, **kwargs):
-        if which_set not in ('train', 'test'):
-            raise ValueError("CIFAR10 only has a train and test set")
-
         kwargs.setdefault('load_in_memory', True)
-        self.which_set = which_set
-
-        super(CIFAR10, self).__init__(self.data_path, which_set,
-                                      **kwargs)
+        super(CIFAR10, self).__init__(self.data_path, which_set, **kwargs)
+        self.default_transformers += (
```

Comment:
Mmm... But is `flatten` really a "default transformer" then? I imagined "default transformers" as being a completely fixed set of transformers for a given dataset, whereas flattening in this case is optional and chosen by the user.

I guess I would still prefer the solution then where CIFAR-10 sets its default transformers as a class attribute, and `H5PYDataset` extends that tuple if necessary in the constructor. But maybe we should reconsider having "flatten" as  keyword argument entirely. It can either be part of the default transformers, or not, but I guess that it's arguably feature envy if we let the dataset choose what transformers will be applied through a keyword argument; it blurs the separation between data loading and transformation that data sets and transformers was supposed to introduce.